pkg/build/nodeimage/internal/kube/builder_bazel.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kube
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/kind/pkg/exec"
+	"sigs.k8s.io/kind/pkg/log"
+)
+
+// bazelBuilder implements Bits for a local Bazel build
+type bazelBuilder struct {
+	kubeRoot string
+	arch     string
+	logger   log.Logger
+}
+
+var _ Builder = &bazelBuilder{}
+
+// NewBazelBuilder returns a new Builder backed by bazel build,
+// given kubeRoot, the path to the kubernetes source directory
+func NewBazelBuilder(logger log.Logger, kubeRoot, arch string) (Builder, error) {
+	return &bazelBuilder{
+		kubeRoot: kubeRoot,
+		arch:     arch,
+		logger:   logger,
+	}, nil
+}
+
+// Build implements Bits.Build
+func (b *bazelBuilder) Build() (Bits, error) {
+	// TODO: don't cd inside the current process
+	// Instead wrap subprocess calls in a shell w/ CD, or something similar
+	// I think windows doesn't work regardless, so using sh should be fine
+	// cd to k8s source
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	// make sure we cd back when done
+	defer func() {
+		// TODO(bentheelder): set return error?
+		_ = os.Chdir(cwd)
+	}()
+	if err := os.Chdir(b.kubeRoot); err != nil {
+		return nil, err
+	}
```

Comment:
this is basically
```
pushd /my/folder/ 
....
popd
```

TIL windows have them https://en.wikipedia.org/wiki/Pushd_and_popd

Reply:
A) this code isn't new.
B) we should eventually stop CD-ing the whole binary
C) those are builtins in a shell, we need to change it in the current process